### PR TITLE
Replaced material ui in FacilityCreate.tsx

### LINKED ISSUE #4952: Replace Material UI components in `src/Components/Facility/FacilityCreate.tsx`:
```tsx
1: import { Card, CardContent } from "@material-ui/core";\n2: import Popover from "@material-ui/core/Popover";
```

### Tasks

Ensure all the below tasks are accomplished if applicable to the file being modified in the PR.

- [ ] Migrate all Material UI components to use new Care UI / custom components. (ref: [🛠️ Tailwinding Care](https://github.com/coronasafe/care_fe/issues/3742))
- [ ] Migrate all Material UI icons to use [🏥 **Care Icons**](https://github.com/coronasafe/care_fe/pull/4071).
- [ ] Migrate `PageTitle` to `Page` component if any. (See [example](https://github.com/coronasafe/care_fe/pull/4796/files#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846))

For more information see the epic issue that tracks this issue.

### PR DESCRIPTION
**Pull request by print-Sathvik**
### <samp>🤖 Generated by Copilot at b549c21</samp>

Improved the user interface and functionality of the `GLocationPicker` component by adding a `Popover` map panel and fixing a close icon bug.

## Proposed Changes

- Fixes #4952 
- Replaced Material UI in FacilityCreate.tsx
- Wrapped close button of GLocationPicker with Popover.Button after replacing material UI's Popover component with headlessui's Popover component as that's the only way to close the Popover according to the docs.
- GLocationPicker component does not load the map in mobile view / at lower width and the search box and close button occupy the entire space as in the below screenshot. Fixed that issue too.

image

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### <samp>🤖 Generated by Copilot at b549c21</samp>

* Import the `Popover` component from `@headlessui/react` library to create a dismissible floating panel for the map ([link](https://github.com/coronasafe/care_fe/pull/5719/files?diff=unified&w=0#diff-c4bc281dcacf25e0ea09e5465988d6b4e3a2aee0b5f9b9a1d804b46e36f73999R8))
* Wrap the close icon for the map in a `Popover.Button` component to control the visibility of the `Popover` component and fix a bug where the map would not close when clicking the icon ([link](https://github.com/coronasafe/care_fe/pull/5719/files?diff=unified&w=0#diff-c4bc281dcacf25e0ea09e5465988d6b4e3a2aee0b5f9b9a1d804b46e36f73999L204-R214))

### CODE REVIEW

File: src/Components/Facility/FacilityCreate.tsx [hunk 1]
```diff
@@ -682,339 +670,350 @@ export const FacilityCreate = (props: FacilityProps) => {
     case 3:
       return (
         <div className="px-2 pb-2">
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: Remove the `div` that's present before `Page`. The page component will add consistent padding for all pages.

File: src/Components/Facility/FacilityCreate.tsx [hunk 2]
```diff
@@ -682,339 +670,350 @@ export const FacilityCreate = (props: FacilityProps) => {
...
+                <div className="font-bold text-xl mb-2">{t("doctors_list")}</div>
+              </div>
+              <div className="mt-4">{doctorList}</div>
+            </div>
+          </Page>
         </div>
       );
     case 2:
       return (
         <div className="px-2 pb-2">
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: Same here

File: src/Components/Facility/FacilityCreate.tsx [hunk 3]
```diff
@@ -682,339 +670,350 @@ export const FacilityCreate = (props: FacilityProps) => {
...
             </div>
-            <div>{capacityList}</div>
-          </div>
+          </Page>
         </div>
       );
     case 1:
     default:
       return (
         <div className="px-2 pb-2">
```
Comment (rithviknishad) [inline src/Components/Facility/FacilityCreate.tsx]: here too

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Rest all LGTM

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@print-Sathvik @rithviknishad 
image

the map is not properly aligned, make it appear near the icon

**Comment by print-Sathvik** [comment]
Can someone else take this up. When I fixed this some other issues starte.

> @print-Sathvik @rithviknishad image
> 
> the map is not properly aligned, make it appear near the icon

**Comment by khavinshankar** [comment]
Fixed the positioning of location picker

image

cc: @nihal467 @print-Sathvik

**Comment by nihal467** [comment]
@print-Sathvik @khavinshankar the cypress is failing, which is related to this PR, fix it

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**